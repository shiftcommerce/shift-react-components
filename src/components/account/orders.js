// Libraries
import React, { Component } from 'react'

// Lib
import componentMapping from '../../lib/component-mapping'

class AccountOrders extends Component {
  constructor (props) {
    super(props)

    this.Button = componentMapping('Button')
    this.Link = componentMapping('Link')
    this.Loading = componentMapping('Loading')
    this.OrderList = componentMapping('OrderList')
  }

  componentDidMount () {
    this.props.fetchOrders()
  }

  renderOrdersList (orders) {
    if (orders.data.length === 0) {
      return (<p>No previous orders found.</p>)
    }

    return (<this.OrderList orders={orders} />)
  }

  renderAccountBanner () {
    return (
      <div className='c-order-history__nav'>
        <h2>Order History</h2>
      </div>
    )
  }

  render () {
    const { orders, orders: { loading } } = this.props

    if (loading) {
      return (
        <>
          <div className='c-order-history'>
            { this.renderAccountBanner() }
            <this.Loading />
          </div>
        </>
      )
    } else {
      return (
        <div className='c-order-history'>
          { this.renderAccountBanner() }
          { this.renderOrdersList(orders) }
        </div>
      )
    }
  }
}

export default AccountOrders
