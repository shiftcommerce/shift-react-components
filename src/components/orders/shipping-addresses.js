// Libraries
import React, { PureComponent } from 'react'

class ShippingAddresses extends PureComponent {
  render () {
    return (
      <>
        { this.props.addresses.map((shippingAddress) => {
          return (
            <div className='c-order-history__shipping-address' key={shippingAddress.id}>
              <p className='u-bold'>{ shippingAddress.name }</p>
              <p>{ shippingAddress.company }</p>
              <p>{ shippingAddress.lines.join() }</p>
              <p>{ shippingAddress.city }</p>
              <p>{ shippingAddress.state }</p>
              <p>{ shippingAddress.postcode }</p>
              <p>{ shippingAddress.country }</p>
            </div>
          )
        })}
      </>
    )
  }
}

export default ShippingAddresses
