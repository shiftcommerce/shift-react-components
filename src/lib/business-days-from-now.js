// Libraries
import addDays from 'date-fns/add_days'
import isWeekend from 'date-fns/is_weekend'

export default function businessDaysFromNow (days) {
  // Convert number of days to integer
  const parsedDays = parseInt(days)

  // Calculate start date
  const startDate = new Date(Date.now())

  // Calculate end date (startDate + parsedDays)
  let endDate = addDays(startDate, parsedDays)

  // Iterate over endDate, increasing by 1 day until it is a business day (Mon - Fri)
  while (isWeekend(endDate)) {
    endDate = addDays(endDate, 1)
  }

  return endDate
}
