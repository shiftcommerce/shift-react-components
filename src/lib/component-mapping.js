import Config from './config'

/**
 * Lib
 */
import FormErrors from './form-errors'

/**
 * Navigation Components
 */
import NavBar from '../components/navigation/navbar'
import NavBarOption from '../components/navigation/navbar-option'

/**
 * Objects
 */
import Breadcrumb from '../objects/breadcrumb'
import Button from '../objects/button'
import Checkbox from '../objects/checkbox'
import ConditionalLink from '../objects/conditional-link'
import DropdownSelect from '../objects/dropdown-select'
import Head from '../objects/head'
import Image from '../objects/image'
import Input from '../objects/input'
import LazyLoad from '../objects/lazy-load'
import Link from '../objects/link'
import Loading from '../objects/loading'
import Logo from '../objects/logo'
import PayPalButton from '../objects/paypal-button'
import Rating from '../objects/rating'
import VariantSelector from '../objects/variant-selector'

/**
 * PDP Components
 */
import ProductCarousel from '../components/products/display/product-carousel'
import ProductEwisForm from '../components/products/display/product-ewis-form'
import ProductPrice from '../components/products/display/product-price'

/**
 * PLP Components
 */
import ProductListingCard from '../components/products/listing/product-listing-card'
import ProductMenu from '../components/products/listing/product-menu'
import ProductMenuOptions from '../components/products/listing/product-menu-options'

/**
 * Cart/Checkout components
 */
import AddressBook from '../components/checkout/address-book'
import AddressForm from '../components/checkout/address-form'
import AddressFormHeader from '../components/checkout/address-form-header'
import LineItems from '../components/cart/line-items'
import PaymentIcons from '../components/cart/payment-icons'
import PaymentMethods from '../components/checkout/payment-methods'
import PaymentMethod from '../components/checkout/payment-method'
import PaymentMethodHeader from '../components/checkout/payment-method-header'
import ShippingMethodsHeader from '../components/checkout/shipping-methods-header'
import StripeFields from '../components/checkout/stripe-fields'
import StripePayment from '../components/checkout/stripe-payment'
import StripeWrapper from '../components/checkout/stripe-wrapper'

/**
 * Order Components
 */
import OrderLineItems from '../components/orders/order-line-items'
import OrderList from '../components/orders/order-list'
import ShippingAddresses from '../components/orders/shipping-addresses'

/**
 * Search Components
 */
import SearchBar from '../components/search/search-bar'
import SearchFilters from '../components/search/search-filters'
import SearchHits from '../components/search/search-hits'
import SearchRatingFilter from '../components/search/search-rating-filter'
import SearchRefinements from '../components/search/search-refinements'
import SearchSlider from '../components/search/search-slider'

/**
 * Static Page
 */
import StaticPageError from '../components/static-page/error'
import StaticPageErrorBody from '../components/static-page/error-body'
import StaticPageErrorDetails from '../components/static-page/error-details'

/**
 * Layout Components
 */
import CustomHead from '../components/layout/custom-head'
import Footer from '../components/layout/footer'
import Minibag from '../components/layout/minibag'

const mapping = {
  AddressBook,
  AddressForm,
  AddressFormHeader,
  Breadcrumb,
  Button,
  Checkbox,
  ConditionalLink,
  CustomHead,
  DropdownSelect,
  Footer,
  FormErrors,
  Head,
  Image,
  Input,
  LazyLoad,
  LineItems,
  Link,
  Loading,
  Logo,
  Minibag,
  NavBar,
  NavBarOption,
  OrderLineItems,
  OrderList,
  PayPalButton,
  PaymentIcons,
  PaymentMethods,
  PaymentMethod,
  PaymentMethodHeader,
  ProductCarousel,
  ProductEwisForm,
  ProductListingCard,
  ProductMenu,
  ProductMenuOptions,
  ProductPrice,
  Rating,
  SearchBar,
  SearchFilters,
  SearchHits,
  SearchRatingFilter,
  SearchRefinements,
  SearchSlider,
  ShippingAddresses,
  ShippingMethodsHeader,
  StaticPageError,
  StaticPageErrorBody,
  StaticPageErrorDetails,
  StripeFields,
  StripePayment,
  StripeWrapper,
  VariantSelector
}

export default function (module) {
  return Config.get()[module] || mapping[module]
}
