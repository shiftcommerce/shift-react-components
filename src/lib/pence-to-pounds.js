/**
 * Convert the price from pence to pounds. For example:
 * 2547 pence becomes 25.47
 * @param  {Number} priceInPence
 * @return {Number} - The price
 */

export function penceToPounds (priceInPence) {
  return (priceInPence / 100).toFixed(2)
}
