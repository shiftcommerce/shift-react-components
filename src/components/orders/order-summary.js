// Libraries
import React, { Component } from 'react'
import Sticky from 'react-stickyfill'

// Lib
import { decimalPrice } from '../../lib/decimal-price'
import componentMapping from '../../lib/component-mapping'

class OrderSummary extends Component {
  constructor (props) {
    super(props)

    this.Image = componentMapping('Image')
    this.Link = componentMapping('Link')
  }

  /**
  * Render line item image for the order summary
  * @param  {Object} lineItem
  * @return {string} - HTML markup for the component
  */
  renderLineItemImage (lineItem) {
    if (lineItem.imageUrl) {
      return (
        <this.Link href={`/slug?slug=${lineItem.slug}`} as={`${lineItem.canonical_path}`} >
          <this.Image src={lineItem.imageUrl} alt={lineItem.title} aria-label={lineItem.title} />
        </this.Link>
      )
    } else {
      return (
        <this.Image src='/static/placeholder.png' alt={lineItem.title} aria-label={lineItem.title} />
      )
    }
  }

  /**
  * Render the line item title for the order summary
  * @param  {Object} lineItem
  * @return {string} - HTML markup for the component
  */
  renderLineItemTitle (lineItem) {
    return (
      <>
        <p className='c-order__summary-line-item-details-title u-bold'>
          { lineItem.title }
        </p>
        <p className='c-order__summary-line-item-details-sku'>
          { lineItem.sku }
        </p>
      </>
    )
  }

  /**
  * Render the line-item meta-data for the order summary
  * @param  {Object} _lineItem
  * @return {string} - HTML markup for the component
  */
  renderLineItemParams (_lineItem) {
    // @TODO: to update this with actual metadata
    const metaData = { colour: 'Grey', size: 'L', quantity: 1 }
    return (
      Object.keys(metaData).map((key, index) =>
        <div className='c-order__summary-line-item-details-param' key={key}>
          <div className='c-order__summary-line-item-details-param-key'>
            <p>{ key }:</p>
          </div>
          <div className='c-order__summary-line-item-details-param-value'>
            <p>{ metaData[key] }</p>
          </div>
        </div>
      )
    )
  }

  /**
  * Render the line items for the order summary
  * @param  {Object} order
  * @return {string} - HTML markup for the component
  */
  renderLineItems (order) {
    return (
      order.line_items.map((lineItem) =>
        <div className='c-order__summary-line-item' key={lineItem.sku}>
          <div className='c-order__summary-line-item-image'>
            { this.renderLineItemImage(lineItem) }
          </div>

          <div className='c-order__summary-line-item-details'>
            { this.renderLineItemTitle(lineItem) }
            { this.renderLineItemParams(lineItem) }
          </div>
        </div>
      )
    )
  }

  /**
  * Render the total prices for the order summary
  * @param  {Object} order
  * @return {string} - HTML markup for the component
  */
  renderOrderTotals (order) {
    return (
      <Sticky>
        <div aria-label='Order total summary' className='u-sticky'>
          <div>
            <dl aria-label='Order subtotal'>
              <dt>Total products:</dt>
              <dd>&pound;{ decimalPrice(order.sub_total) }</dd>
            </dl>
            <dl aria-label='Order shipping costs'>
              <dt>Shipping costs:</dt>
              <dd>&pound;{ order.shipping_total }</dd>
            </dl>
            <dl aria-label='Order total' className='u-bold'>
              <dt>TOTAL:</dt>
              <dd>&pound;{ decimalPrice(order.total) }</dd>
            </dl>
            <dl>
              <dt>* Including VAT</dt>
            </dl>
          </div>
        </div>
      </Sticky>
    )
  }

  render () {
    const { order } = this.props

    return (
      <div className='c-order__summary'>
        <div className='c-order__summary-header'>
          <div className='c-order__summary-header-title'>
            <h2>Your Order</h2>
          </div>

          <div className='c-order__summary-header-total'>
            <h2>&pound;{ decimalPrice(order.total) }</h2>
          </div>
        </div>

        { this.renderLineItems(order) }

        <div className='c-order__summary-totals'>
          { this.renderOrderTotals(order) }
        </div>
      </div>
    )
  }
}

export default OrderSummary
