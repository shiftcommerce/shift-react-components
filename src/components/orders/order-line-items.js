// Libraries
import React, { PureComponent } from 'react'
import t from 'typy'

// Lib
import { decimalPrice } from '../../lib/decimal-price'
import { penceToPounds } from '../../lib/pence-to-pounds'
import componentMapping from '../../lib/component-mapping'

class OrderLineItems extends PureComponent {
  constructor (props) {
    super(props)

    this.Image = componentMapping('Image')
    this.Link = componentMapping('Link')
  }

  /**
  * Render the total prices for the line item
  * @param  {Object} lineItem
  * @return {string} - HTML markup for the component
  */
  renderTotal (lineItem) {
    const total = t(lineItem, 'pricing.line_total_inc_tax').safeObject
    const formattedTotal = decimalPrice(total)

    return (
      <div className='c-order-history__line-items_grid_item2'>
        <a>Â£{ penceToPounds(formattedTotal) }</a>
      </div>
    )
  }

  /**
  * Render the title & sku for the line item
  * @param  {Object} lineItem
  * @return {string} - HTML markup for the component
  */
  renderTitle (lineItem) {
    // To be updated, see: https://github.com/shiftcommerce/shift-front-end-react/issues/713
    return (
      <>
        <h5>{ lineItem.sku }</h5>
        <h4>{ lineItem.sku }</h4>
      </>
    )
  }

  /**
  * Render the quantity, delivery method & customer name for the line item
  * @param  {Object} lineItem
  * @return {string} - HTML markup for the component
  */
  renderParams (lineItem) {
    const deliveryMethod = t(lineItem, 'shipping_method.label').safeObject
    const customerName = t(lineItem, 'shipping_address.name').safeObject

    return (
      <div className='c-order-history__line-items_grid_item1'>
        { this.renderTitle(lineItem) }
        <span>Quantity: { lineItem.quantity }</span>
        <span>Delivery: { deliveryMethod }</span>
        <span>Address: { customerName }</span>
      </div>
    )
  }

  renderLineItems () {
    const { items } = this.props

    return (
      items.map((lineItem) =>
        <div className='c-order-history__line-items_grid' key={lineItem.sku}>
          { this.renderParams(lineItem) }
          { this.renderTotal(lineItem) }
        </div>
      )
    )
  }

  render () {
    return this.renderLineItems()
  }
}

export default OrderLineItems
