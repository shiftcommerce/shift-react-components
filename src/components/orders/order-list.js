// Libraries
import React, { PureComponent } from 'react'
import dateFns from 'date-fns'
import t from 'typy'

// Lib
import { penceToPounds } from '../../lib/pence-to-pounds'
import componentMapping from '../../lib/component-mapping'

class OrderList extends PureComponent {
  constructor (props) {
    super(props)

    this.OrderLineItems = componentMapping('OrderLineItems')
    this.ShippingAddresses = componentMapping('ShippingAddresses')
  }

  /**
  * Render the total shipping costs in the order list
  * @param  {Object} order
  * @return {string} - HTML markup for the component
  */
  renderShippingTotal (order) {
    const shippingTotal = order.shipping_methods.reduce(
      (accumulator, currentValue) => accumulator + currentValue.price, 0
    )

    return penceToPounds(shippingTotal)
  }

  /**
  * Render the total prices for the line item
  * @param  {Object} order
  * @param  {Object} orderDate
  * @param  {Object} total
  * @return {string} - HTML markup for the component
  */
  renderOrderSummary (order, orderDate, total) {
    return (
      <div className='c-order-history__order-summary'>
        <div className='c-order-history__order-summary_params'>
          <p className='u-bold u-inline-block'>Date of Order: </p>
          <a className='u-inline-block'>{ orderDate }</a>
        </div>
        <div className='c-order-history__order-summary_params'>
          <p className='u-bold u-inline-block'>Items: </p>
          <a className='u-inline-block'>{ order.line_items.length }</a>
        </div>
        <div className='c-order-history__order-summary_params'>
          <p className='u-bold u-inline-block'>Order Number: </p>
          <a className='u-inline-block'>{ order.reference }</a>
        </div>
        <div className='c-order-history__order-summary_params'>
          <p className='u-bold u-inline-block'>Total: </p>
          <a className='u-inline-block'>&pound;{ total }</a>
        </div>
      </div>
    )
  }

  /**
  * Render the order details
  * @param  {Object} order
  * @param  {Object} orderDate
  * @param  {Object} total
  * @return {string} - HTML markup for the component
  */
  renderOrderDetails (order, orderDate, total) {
    return (
      <div className='c-order-history__details'>
        { this.renderOrderSummary(order, orderDate, total) }
        <this.OrderLineItems items={order.line_items} />
        <div className='c-order-history__totals'>
          <p>Shipping: &pound;{ this.renderShippingTotal(order) } </p>
          <p className='u-bold'>Total: &pound;{ total }</p>
        </div>
        <h3>Shipping Details</h3>
        <this.ShippingAddresses addresses={order.shipping_addresses} />
      </div>
    )
  }

  /**
  * Render the header for the order list
  * @param  {Object} order
  * @param  {Object} orderDate
  * @param  {Object} total
  * @return {string} - HTML markup for the component
  */
  renderOrderHeader (order, orderDate, total) {
    return (
      <>
        <h4>{ orderDate } - { order.reference } - Â£{ total }</h4>
        <label htmlFor={order.reference} className='c-order-history__header-label' />
        <input type='checkbox' id={order.reference} />
        <span className='c-order-history__arrow' />
      </>
    )
  }

  render () {
    const { orders } = this.props

    return (
      <>
        {orders.data.map((order) => {
          const orderDate = dateFns.format(new Date(order.placed_at), 'MMM D, YYYY')
          const total = penceToPounds(t(order, 'pricing.total_inc_tax').safeObject)

          return (
            <div className='c-order-history__header' key={order.id}>
              { this.renderOrderHeader(order, orderDate, total) }
              { this.renderOrderDetails(order, orderDate, total) }
            </div>
          )
        })}
      </>
    )
  }
}

export default OrderList
