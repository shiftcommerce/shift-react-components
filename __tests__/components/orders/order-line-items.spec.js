// Libraries
import React from 'react'

// Component
import OrderLineItems from '../../../src/components/orders/order-line-items'

// Lib
import { penceToPounds } from '../../../src/lib/pence-to-pounds'

// Fixtures
import orders from '../../fixtures/orders'

test('renders correctly', () => {
  // Arrange
  const lineItems = orders.data[0].line_items

  // Act
  const wrapper = mount(
    <OrderLineItems items={lineItems} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(lineItems[0].sku)
  expect(wrapper).toIncludeText(lineItems[0].quantity)
  expect(wrapper).toIncludeText(penceToPounds(lineItems[0].pricing.line_inc_tax))
  expect(wrapper.find('h5').at(1)).toIncludeText(lineItems[1].sku)
  expect(wrapper.find('span').at(3)).toIncludeText(lineItems[1].quantity)
  expect(wrapper.find('a').at(1)).toIncludeText(penceToPounds(lineItems[1].pricing.line_inc_tax))
})
