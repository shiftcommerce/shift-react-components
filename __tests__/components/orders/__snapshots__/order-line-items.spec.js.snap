// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<OrderLineItems
  items={
    Array [
      Object {
        "discounts": Array [],
        "id": "61c40964-4b4c-4f54-b580-958798087312_0",
        "image_urls": Array [
          "https://example.com/path/to/image.jpg",
          "https://example.com/path/to/image1.jpg",
        ],
        "individual_prices": Array [],
        "pricing": Object {
          "each_ex_tax": 1200,
          "each_inc_tax": 1440,
          "line_discount_ex_tax": 0,
          "line_discount_inc_tax": 0,
          "line_ex_tax": 6000,
          "line_total_inc_tax": 7200,
          "tax_rate": 12000,
        },
        "quantity": 5,
        "shipping_address": Object {
          "city": "Leeds",
          "company": "SHIFT Commerce",
          "country": "United Kingdom",
          "id": "61c40964-4b4c-4f54-b580-958798087312_1",
          "lines": Array [
            "The Calls",
          ],
          "meta_attributes": Object {
            "further": "information",
          },
          "name": "Test Customer",
          "postcode": "LS2 7EY",
          "state": "West Yorkshire",
        },
        "shipping_method": Object {
          "id": "61c40964-4b4c-4f54-b580-958798087312_1",
          "label": "Next Day Delivery",
          "meta_attributes": Object {
            "sku": "NEXTDAY",
          },
          "price": 599,
        },
        "sku": "WHITE-TSHIRT-XXL",
      },
      Object {
        "discounts": Array [
          Object {
            "amount_ex_tax": 200,
            "amount_inc_tax": 240,
            "coupon_code": "3FOR2SOCKS",
            "id": "61c40964-4b4c-4f54-b580-958798087312_1_0",
            "label": "3 for 2 on Socks",
            "meta_attributes": Object {},
          },
        ],
        "id": "61c40964-4b4c-4f54-b580-958798087312_1",
        "image_urls": Array [],
        "individual_prices": Array [
          Object {
            "discount_ex_tax": 67,
            "discount_inc_tax": 80,
            "discounts": Array [
              Object {
                "amount_ex_tax": 67,
                "amount_inc_tax": 80,
                "coupon_code": null,
                "id": "61c40964-4b4c-4f54-b580-958798087312_1_0_0",
                "label": "3 for 2 on Socks",
                "meta_attributes": Object {},
              },
            ],
            "ex_tax": 133,
            "id": "61c40964-4b4c-4f54-b580-958798087312_1_0",
            "inc_tax": 160,
          },
          Object {
            "discount_ex_tax": 67,
            "discount_inc_tax": 80,
            "discounts": Array [
              Object {
                "amount_ex_tax": 67,
                "amount_inc_tax": 80,
                "coupon_code": null,
                "id": "61c40964-4b4c-4f54-b580-958798087312_1_1_0",
                "label": "3 for 2 on Socks",
                "meta_attributes": Object {},
              },
            ],
            "ex_tax": 133,
            "id": "61c40964-4b4c-4f54-b580-958798087312_1_1",
            "inc_tax": 160,
          },
          Object {
            "discount_ex_tax": 66,
            "discount_inc_tax": 80,
            "discounts": Array [
              Object {
                "amount_ex_tax": 66,
                "amount_inc_tax": 80,
                "coupon_code": null,
                "id": "61c40964-4b4c-4f54-b580-958798087312_1_2_0",
                "label": "3 for 2 on Socks",
                "meta_attributes": Object {},
              },
            ],
            "ex_tax": 134,
            "id": "61c40964-4b4c-4f54-b580-958798087312_1_2",
            "inc_tax": 160,
          },
        ],
        "pricing": Object {
          "each_ex_tax": 200,
          "each_inc_tax": 240,
          "line_discount_ex_tax": 200,
          "line_discount_inc_tax": 240,
          "line_ex_tax": 400,
          "line_total_inc_tax": 480,
          "tax_rate": 12000,
        },
        "quantity": 3,
        "shipping_address": Object {
          "city": "Leeds",
          "company": null,
          "country": "United Kingdom",
          "id": "61c40964-4b4c-4f54-b580-958798087312_2",
          "lines": Array [
            "SHIFT Commerce",
            "The Calls Landing",
          ],
          "meta_attributes": Object {
            "further": "information",
          },
          "name": "Test Customer 2",
          "postcode": "LS2 7EY",
          "state": "West Yorkshire",
        },
        "shipping_method": Object {
          "id": "61c40964-4b4c-4f54-b580-958798087312_2",
          "label": "Two-man Delivery",
          "meta_attributes": Object {
            "sku": "DROPSHIP",
          },
          "price": 2499,
        },
        "sku": "BLACK-SOCKS-001",
      },
      Object {
        "discounts": Array [],
        "id": "61c40964-4b4c-4f54-b580-958798087312_2",
        "image_urls": Array [],
        "individual_prices": Array [],
        "pricing": Object {
          "each_ex_tax": 120000,
          "each_inc_tax": 144000,
          "line_discount_ex_tax": 0,
          "line_discount_inc_tax": 0,
          "line_ex_tax": 120000,
          "line_total_inc_tax": 144000,
          "tax_rate": 12000,
        },
        "quantity": 1,
        "shipping_address": Object {
          "city": "Leeds",
          "company": "SHIFT Commerce",
          "country": "United Kingdom",
          "id": "61c40964-4b4c-4f54-b580-958798087312_1",
          "lines": Array [
            "The Calls",
          ],
          "meta_attributes": Object {
            "further": "information",
          },
          "name": "Test Customer",
          "postcode": "LS2 7EY",
          "state": "West Yorkshire",
        },
        "shipping_method": Object {
          "id": "61c40964-4b4c-4f54-b580-958798087312_1",
          "label": "Next Day Delivery",
          "meta_attributes": Object {
            "sku": "NEXTDAY",
          },
          "price": 599,
        },
        "sku": "WHITE-TSHIRT-L",
      },
    ]
  }
>
  <div
    className="c-order-history__line-items_grid"
    key="WHITE-TSHIRT-XXL"
  >
    <div
      className="c-order-history__line-items_grid_item1"
    >
      <h5>
        WHITE-TSHIRT-XXL
      </h5>
      <h4>
        WHITE-TSHIRT-XXL
      </h4>
      <span>
        Quantity: 
        5
      </span>
      <span>
        Delivery: 
        Next Day Delivery
      </span>
      <span>
        Address: 
        Test Customer
      </span>
    </div>
    <div
      className="c-order-history__line-items_grid_item2"
    >
      <a>
        £
        72.00
      </a>
    </div>
  </div>
  <div
    className="c-order-history__line-items_grid"
    key="BLACK-SOCKS-001"
  >
    <div
      className="c-order-history__line-items_grid_item1"
    >
      <h5>
        BLACK-SOCKS-001
      </h5>
      <h4>
        BLACK-SOCKS-001
      </h4>
      <span>
        Quantity: 
        3
      </span>
      <span>
        Delivery: 
        Two-man Delivery
      </span>
      <span>
        Address: 
        Test Customer 2
      </span>
    </div>
    <div
      className="c-order-history__line-items_grid_item2"
    >
      <a>
        £
        4.80
      </a>
    </div>
  </div>
  <div
    className="c-order-history__line-items_grid"
    key="WHITE-TSHIRT-L"
  >
    <div
      className="c-order-history__line-items_grid_item1"
    >
      <h5>
        WHITE-TSHIRT-L
      </h5>
      <h4>
        WHITE-TSHIRT-L
      </h4>
      <span>
        Quantity: 
        1
      </span>
      <span>
        Delivery: 
        Next Day Delivery
      </span>
      <span>
        Address: 
        Test Customer
      </span>
    </div>
    <div
      className="c-order-history__line-items_grid_item2"
    >
      <a>
        £
        1440.00
      </a>
    </div>
  </div>
</OrderLineItems>
`;
